### PR TITLE
Cluster-Config and Dataloader

### PR DESCRIPTION
**Pull request by prithvi081099**

### CODE REVIEW

File: gateway/modules/crud/helpers.go [hunk 1]
```diff
@@ -68,3 +68,7 @@ func splitConnectionString(connection string) (string, bool) {
 	}
 	return "", false
 }
+
+func (m *Module) getIsDataloader(dbAlias string) bool {
```
Comment (sharadregoti) [inline gateway/modules/crud/helpers.go:72]:
```suggestion
func (m *Module) isRequestBatchable(isBatch bool,dbAlias string) bool {
```

Comment (sharadregoti) [reply gateway/modules/crud/helpers.go:72]: Also, verify if the DB alias is same, if not then return false.

File: space-cli/cmd/modules/cluster/getters.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// GetIntegration gets integration
+func GetIntegration() ([]*model.SpecObject, error) {
+	url := fmt.Sprintf("/v1/config/integrations")
+
+	// Get the spec from the server
+	payload := new(model.Response)
+	if err := transport.Client.MakeHTTPRequest(http.MethodGet, url, map[string]string{}, payload); err != nil {
+		return nil, err
+	}
+
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters.go:45]: Implement the remaining

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: gateway/config/config.go [hunk 3]
```diff
@@ -103,6 +103,7 @@ type DatabaseConfig struct {
 	DBName       string       `json:"name,omitempty" yaml:"name" mapstructure:"name"` // name of the logical database or schema name according to the database type
 	Conn         string       `json:"conn,omitempty" yaml:"conn" mapstructure:"conn"`
 	IsPrimary    bool         `json:"isPrimary" yaml:"isPrimary" mapstructure:"isPrimary"`
+	IsDataloader bool         `json:"isDataloader" yaml:"isDataloader" mapstructure:"isDataloader"`
```
Comment (sharadregoti) [inline gateway/config/config.go:106]:
```suggestion
	IsDataloaderEnabled bool         `json:"isDataloader" yaml:"isDataloader" mapstructure:"isDataloader"`
```

File: space-cli/cmd/modules/cluster/getters.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/spaceuptech/space-cloud/space-cli/cmd/utils"
+	"github.com/spaceuptech/space-cloud/space-cli/cmd/utils/transport"
+)
+
+// GetClusterConfig gets clusters config
+func GetClusterConfig() ([]*model.SpecObject, error) {
+	url := fmt.Sprintf("/v1/config/cluster")
+
+	// Resp is the object returned by every handler to client
+	type Resp struct {
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters.go:17]:
```suggestion
	type resp struct {
```
Remove the comment or change it

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Comment by sharadregoti** [comment]
@prithvi081099 Add test cases for the new commands that you have added in space cli.

File: space-cli/cmd/modules/cluster/getters_test.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+		args           []interface{}
+		paramsReturned []interface{}
+	}
+	tests := []struct {
+		name              string
+		transportMockArgs []mockArgs
+		want              []*model.SpecObject
+		wantErr           bool
+	}{
+		// TODO: Add test cases.
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters_test.go:105]: Remove this

File: space-cli/cmd/modules/cluster/getters_test.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters_test.go:24]: remove this

File: space-cli/cmd/modules/cluster/getters_test.go [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+		{
+			name: "Successful test",
+			transportMockArgs: []mockArgs{
+				{
+					method: "MakeHTTPRequest",
+					args:   []interface{}{"GET", "/v1/config/cluster", map[string]string{}, new(resp)},
+					paramsReturned: []interface{}{nil, resp{
+						Result: map[string]interface{}{
+							"letsEncryptEmail": "",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters_test.go:33]:
```suggestion
							"letsEncryptEmail": "info@gmail.com",
```

File: space-cli/cmd/modules/cluster/getters_test.go [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+						},
+					}},
+				},
+			},
+			want: []*model.SpecObject{
+				{
+					API:  "/v1/config/cluster",
+					Type: "cluster-config",
+					Meta: map[string]string{},
+					Spec: map[string]interface{}{"clusterConfig": map[string]interface{}{"letsEncryptEmail": "", "enableTelemetry": false}},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters_test.go:44]:
```suggestion
					Spec: map[string]interface{}{"clusterConfig": map[string]interface{}{"letsEncryptEmail": "info@gmail.com", "enableTelemetry": false}},
```

File: space-cli/cmd/modules/cluster/getters_test.go [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Successful test",
+			transportMockArgs: []mockArgs{
+				{
+					method: "MakeHTTPRequest",
+					args:   []interface{}{"GET", "/v1/config/integrations", map[string]string{}, new(model.Response)},
+					paramsReturned: []interface{}{nil, model.Response{
+						Result: []interface{}{map[string]interface{}{}},
```
Comment (sharadregoti) [inline space-cli/cmd/modules/cluster/getters_test.go:113]: The result cannot be empty, we have to provide values

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Review by sharadregoti — DISMISSED** [review]

**Review by YourTechBud — CHANGES_REQUESTED** [review]
Make this request level not database level

**Comment by sharadregoti** [comment]
Fixes #1474 & #1484

File: space-cli/cmd/modules/cluster/commands.go [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+func GetSubCommands() []*cobra.Command {
+	var getClusterConfig = &cobra.Command{
+		Use:     "cluster-config",
+		Aliases: []string{"cluster-configs"},
+		RunE:    actionGetClusterConfig,
+	}
+
+	var getIntegration = &cobra.Command{
+		Use:     "integration",
+		Aliases: []string{"integrationss"},
```
Comment (YourTechBud) [inline space-cli/cmd/modules/cluster/commands.go:18]:
```suggestion
		Aliases: []string{"integrations"},
```

File: space-cli/cmd/modules/cluster/getters.go [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+	payload := new(model.Response)
+	if err := transport.Client.MakeHTTPRequest(http.MethodGet, url, map[string]string{}, payload); err != nil {
+		return nil, err
+	}
+
+	var objs []*model.SpecObject
+
+	for _, item := range payload.Result {
+		spec := item.(map[string]interface{})
+		meta := map[string]string{}
```
Comment (YourTechBud) [inline space-cli/cmd/modules/cluster/getters.go:55]: Shouldn't we store the id of the integration in the `meta`? Would we need that while applying?

**Review by YourTechBud — CHANGES_REQUESTED** [review]

File: space-cli/cmd/modules/cluster/getters.go (hunk 11 above)
Comment (sharadregoti) [reply space-cli/cmd/modules/cluster/getters.go:55]: Whatever we store in meta is used, for filling the values in the URL.
But in the URL of integration, there is no ID param. That's why we are not storing id in the meta

File: space-cli/cmd/modules/operations/apply.go [hunk 12]
```diff
@@ -92,7 +92,7 @@ func ApplySpec(token string, account *model.Account, specObj *model.SpecObject)
 	if err != nil {
 		return err
 	}
-
+	fmt.Println("Details", string(requestBody), url)
```
Comment (YourTechBud) [inline space-cli/cmd/modules/operations/apply.go:95]:
```suggestion

```

**Review by YourTechBud — CHANGES_REQUESTED** [review]

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**